addons/fire/functions/fnc_getScreams.sqf

```diff
@@ -0,0 +1,36 @@
+#include "..\script_component.hpp"
+/*
+ * Author: DartRuffian
+ * Returns a list of scream sounds that a unit will play when on fire.
+ *
+ * Arguments:
+ * 0: Unit <OBJECT|STRING>
+ *
+ * Return Value:
+ * Scream sounds <ARRAY of STRING>
+ *
+ * Example:
+ * player call ace_fire_fnc_getScreams
+ *
+ * Public: Yes
+ */
+
+params [["_unitClass", objNull, [objNull, ""]]];
+TRACE_1("fnc_getScreams",_unitClass);
+
+if (_unitClass isEqualType objNull) then {
+    _unitClass = typeOf _unitClass;
+};
+if (_unitClass == "") exitWith {[]};
+
+// If unit is defined in hash, grab sounds and return
+// If not, check each parent of the class until a value is defined
+GVAR(screams) getOrDefaultCall [_unitClass, {
+    private _type = _unitClass;
+    private _return = [];
+    while {true} do {
+        _type = configName inheritsFrom (configFile >> "CfgVehicles" >> _type);
```

Comment:
This is an infinite loop if no parent is in the list.
This can occur, as you don't check if the input actually inherits from `CAManBase`.